src/de.rs

```diff
@@ -571,6 +563,14 @@ where
     where
         V: de::Visitor<'de>,
     {
+        // Peek to check if we have a tag. If so, we tell the visitor to visit a newtype.
+        //
+        // They'll then call us back with the specific newtype they're looking for, and it better be
+        // a CID.
+        if self.peek_is_tag()? {
+            return visitor.visit_newtype_struct(self);
```

Comment:
Do we need checked recursion here?

Reply:
I'd say in our specific case not, as we only support one specific non-recursive tag. We could add a check for consistency though.